codec-http2/src/main/java/io/netty/handler/codec/http2/ClearTextHttp2ServerDispatcher.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.http2;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufUtil;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandler;
+import io.netty.handler.codec.ByteToMessageDecoder;
+import io.netty.util.AsciiString;
+import io.netty.util.internal.UnstableApi;
+
+import java.util.List;
+
+import static io.netty.buffer.Unpooled.*;
+
+/**
+ * Provide how starting http2 protocol by h2c (Upgrade) or h2 (Prior Knowledge) by peaking the
+ * first bytes is prior knowledge or not, see {@link #PRI_KNOWLEDGE_FIRST_BYTES}
+ * The dispatcher should be the first {@link ChannelInboundHandler} in the
+ * {@link io.netty.channel.ChannelPipeline}
+ * The msg from {@link #channelRead(ChannelHandlerContext, Object)} must be {@link ByteBuf}
+ */
+@UnstableApi
+public abstract class ClearTextHttp2ServerDispatcher extends ByteToMessageDecoder {
+    private static final AsciiString PRI_KNOWLEDGE_MAGIC = AsciiString.of("PRI");
```

Comment:
why not use the full prior knowledge string?  I think it should be "PRI *
   HTTP/2.0\r\n\r\nSM\r\n\r\n" from the http/2 spec.